pkg/controller/provider/container/vsphere/model.go

```diff
@@ -613,6 +613,26 @@ func (v *VmAdapter) Apply(u types.ObjectUpdate) {
 						}
 					}
 				}
+			case fGuestNet:
+				if nics, cast := p.Val.(types.ArrayOfGuestNicInfo); cast {
+					guestNetworksList := []model.GuestNetwork{}
+					for _, info := range nics.GuestNicInfo {
+						if info.IpConfig == nil {
+							continue
+						}
+						ipConfigIp := &info.IpConfig.IpAddress
+						for _, ip := range *ipConfigIp {
+							guestNetworksList = append(guestNetworksList, model.GuestNetwork{
+								MAC:    info.MacAddress,
+								IP:     ip.IpAddress,
+								Origin: ip.Origin,
+							})
+						}
+					}
+					if len(guestNetworksList) > 0 {
```

Comment:
this condition is because once the vm is shutdown, we get an update of empty values.
maybe we can make a better condition. as now if the user remove all his networks in the vm, we won't let it here.

Reply:
it's not that likely that someone would remove all the networks but even if that happens, I don't see an issue in keeping the mac to ip-address set, do you see a problem with this?

and if it was important enough to add this comment here, then it should also be on the source code

Reply:
i didn't try to run virt-v2v with non-existing MAC on the guest, i think it will just ignore it, in that case we are good.

Reply:
tested - works